backend/base/updater/updater.go

```diff
@@ -0,0 +1,108 @@
+package updater
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"time"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/logging"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable/action"
+
+	"github.com/gorilla/websocket"
+	"github.com/sirupsen/logrus"
+)
+
+// Updater implements observable blockchainInfo.
+type Updater struct {
+	observable.Implementation
+	blockchainInfo string
+	log            *logrus.Entry
+	running        bool
+}
+
+// BlockInfo returns the last received blockchain information packet from the middleware
+func (updater *Updater) BlockInfo() string {
+	return updater.blockchainInfo
+}
+
+// NewUpdater returns a new base updater.
+func NewUpdater() *Updater {
+	updater := &Updater{
+		log: logging.Get().WithGroup("base"),
+	}
+	return updater
+}
+
+//Connect start the websocket go routine, first checking if the middleware is reachable
+func (updater *Updater) Connect(ip string, bitboxBaseID string) error {
+	//TODO: This is where the initial rest call should go. The initial call initiates the noise authentication pairing.
+	//This should not be asynchronous, the frontend should not show something until the pairing initiates
+	simpleURL := url.URL{Scheme: "http", Host: ip, Path: "/"}
+	response, err := http.Get(simpleURL.String())
+	//defer response.Body.Close()
+	if err != nil {
+		updater.log.Println("No response from middleware", err)
+		updater.running = false
+		return err
+	}
+	if response.StatusCode == http.StatusOK {
```

Comment:
to save an indentation level and have consistent code flow, I prefer to handle the error first:

```
if response.StatusCode != http.StatusOK {
    return errp.New("...")
}
// continue with success handling
```